lib/modules/asset/AssetService.ts

```diff
@@ -304,213 +303,105 @@ export class AssetService {
+          // Create the assets in the new tenant, with empty linkedDevices and groups
+          const assetContent = Object.assign({}, asset._source);
+          assetContent.linkedDevices = [];
+          assetContent.groups = [];
+          await this.sdk.document.create<AssetContent>(
+            newEngineId,
+            InternalCollection.ASSETS,
+            assetContent,
+            asset._id
           );
-        }
-        const assetsToMigrate = assets.successes.map((asset) => ({
-          _id: asset._id,
-          body: asset._source,
-        }));
 
-        const devices = await this.sdk.document.search<AssetContent>(
-          engineId,
-          InternalCollection.DEVICES,
-          {
-            query: {
-              bool: {
-                filter: {
-                  terms: {
-                    assetId: assetsList,
-                  },
-                },
-              },
-            },
-          }
-        );
-
-        // Map linked devices for assets.
-        const assetLinkedDevices = assets.successes
-          .filter((asset) => asset._source.linkedDevices.length > 0)
-          .map((asset) => ({
-            assetId: asset._id,
-            linkedDevices: asset._source.linkedDevices,
+          // get linked devices to this asset, if any
+          const linkedDevices = asset._source.linkedDevices.map((d) => ({
+            id: d._id,
+            measureNames: d.measureNames,
           }));
 
-        // Extra recovery step to relink back assets to their devices in case of rollback
-        recovery.addRecovery(async () => {
-          // Link the devices to the new assets
-          const promises: Promise<void>[] = [];
-
-          for (const asset of assetLinkedDevices) {
-            const assetId = asset.assetId;
-            for (const device of asset.linkedDevices) {
-              const deviceId = device._id;
-              const measureNames = device.measureNames;
-              promises.push(
-                ask<AskDeviceLinkAsset>(
-                  "ask:device-manager:device:link-asset",
-                  {
-                    assetId,
-                    deviceId,
-                    engineId,
-                    measureNames: measureNames,
-                    user,
-                  }
-                )
-              );
-            }
-          }
-          await Promise.all(promises);
-        });
-
-        // detach from current tenant
-        await Promise.all(
-          devices.hits.map((device) => {
-            return ask<AskDeviceDetachEngine>(
+          // ... ant iterate over this list
+          for (const device of linkedDevices) {
+            // detach linked devices from current tenant (it also unkinks asset)
+            await ask<AskDeviceDetachEngine>(
               "ask:device-manager:device:detach-engine",
-              { deviceId: device._id, user }
+              { deviceId: device.id, user }
```

Comment:
can you keep this property to _id to be coherent with the document content

Reply:
yes, I ll modify it.